### ISSUE TITLE
Should "extends @company/browserslist-config/desktop" work?

### ISSUE THREAD

**Issue by Schweinepriester** [browserslist/browserslist#284]
Hi!

We have published our browserslist-config as `@company/browserslist-config` and are now trying to reference specific files:

```json
"browserslist": [
  "extends @company/browserslist-config/desktop",
  "extends @company/browserslist-config/mobile"
],
```

```bash
[farmer@2c0975f0e531 app]$ npm ls browserslist
…
`-- autoprefixer@9.1.0
  `-- browserslist@4.0.1
```

This fails:

```bash
[11:54:51] 'autoprefixer' errored after 155 ms
[11:54:51] BrowserslistError in plugin "gulp-postcss"
Message:
    Browserslist config needs `browserslist-config-` prefix.  Use `dangerousExtend` option to disable.
Details:
    browserslist: true
    fileName: /app/src/qbShiny.scss
    domainEmitter: [object Object]
    domain: [object Object]
    domainThrown: false

Stack:
BrowserslistError: Browserslist config needs `browserslist-config-` prefix.  Use `dangerousExtend` option to disable.
    at checkExtend (/app/node_modules/browserslist/node.js:21:11)
    at Object.loadQueries (/app/node_modules/browserslist/node.js:116:35)
    at Function.select (/app/node_modules/browserslist/index.js:735:26)
    at /app/node_modules/browserslist/index.js:141:33
    at Array.reduce (<anonymous>)
    at resolve (/app/node_modules/browserslist/index.js:122:18)
    at browserslist (/app/node_modules/browserslist/index.js:219:16)
    at Browsers.parse (/app/node_modules/autoprefixer/lib/browsers.js:65:12)
    at new Browsers (/app/node_modules/autoprefixer/lib/browsers.js:50:26)
    at loadPrefixes (/app/node_modules/autoprefixer/lib/autoprefixer.js:78:20)
```

Without referencing works:

```json
"browserslist": [
    "extends @company/browserslist-config"
  ],
```

==> Should `"extends @company/browserslist-config/desktop"` work?

---

Imho <https://github.com/browserslist/browserslist#shareable-configs> doesn't state that scoped packages and referencing specific files is mutually exclusive.

Thus I'd argue that `SCOPED_CONFIG__PATTERN` at https://github.com/browserslist/browserslist/blob/master/node.js#L9 is too strict and should be something like

```js
var SCOPED_CONFIG__PATTERN = /@[^./]+\/browserslist-config(-|$|\/)/ // added `|\/`
```

or the docs need adjustment.

For both options I could create a pull request, if you want me to.

What do you think? Or might I be missing something?

Thanks! :)

**Comment by ai** [comment]
Hm, good idea. Do you want to send PR?

**Comment by Schweinepriester** [comment]
Sure!

For the record, adding `|\/` also allows `@company/browserslist-config/` (trailing `/`), but that doesn't seem to be a problem.

**Comment by ai** [comment]
Released in 4.3
